parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputFormat.java

```diff
@@ -253,7 +283,23 @@ public ParquetInputFormat() {
   @Override
   public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
     Configuration configuration = ContextUtil.getConfiguration(jobContext);
-    return new ArrayList<InputSplit>(getSplits(configuration, getFooters(jobContext)));
+    List<InputSplit> splits = new ArrayList<InputSplit>();
+
+    if (isTaskSideMetaData(configuration)) {
```

Comment:
this whole block could replace the content of TaskSideMetadataSplitStrategy. Why is it not? It is a little weird to keep the ClientSideMetadataSplitStrategy but not the other one. Either we keep the strategy way of organizing the code or we don't but it should not be in the middle.